src/Features/CSharp/Portable/ConvertForToForEach/CSharpConvertForToForEachCodeRefactoringProvider.cs

```diff
@@ -38,9 +38,10 @@ protected override SyntaxList<StatementSyntax> GetBodyStatements(ForStatementSyn
 
         protected override bool TryGetForStatementComponents(
             ForStatementSyntax forStatement,
-            out SyntaxToken iterationVariable, out ExpressionSyntax initializer,
-            out MemberAccessExpressionSyntax memberAccess,
-            out ExpressionSyntax stepValueExpressionOpt,
+            out SyntaxToken iterationVariable,
+            [NotNullWhen(true)] out ExpressionSyntax? initializer,
+            [NotNullWhen(true)] out MemberAccessExpressionSyntax? memberAccess,
+            [NotNullWhen(true)] out ExpressionSyntax? stepValueExpressionOpt,
```

Comment:
❗ This is not a correct annotation.

Reply:
indeed.  fixed.  not sure how this slipped through either.  @RikkiGibson ?

Reply:
It seems like we should flow this through.  'Try' methods deferring to another 'Try' method is a really simple and normal pattern to encounter:-)